include/NeoFOAM/mesh/unstructuredMesh/BoundaryMesh.hpp

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include <array>
+#include <vector>
+
+#include "NeoFOAM/fields/FieldTypeDefs.hpp"
+#include "NeoFOAM/primitives/label.hpp"
+
+namespace NeoFOAM
+{
+
+class BoundaryMesh
+{
+public:
+
+    BoundaryMesh(
+        const executor& exec,
+        labelField faceCells,
+        vectorField Cf,
+        vectorField Cn,
+        vectorField Sf,
+        scalarField magSf,
+        vectorField nf,
+        vectorField delta,
+        scalarField weights,
+        scalarField deltaCoeffs,
+        std::vector<localIdx> offset
+    );
+
+
+    const labelField& faceCells() const;
+
+    std::span<const label> faceCells(const int i) const;
+
+    const vectorField& Cf() const;
+
+    std::span<const Vector> Cf(const int i) const;
```

Comment:
For compatibility we should keep the naming close to OpenFOAM. But from the name it self it is not clear if it is a simple getter or if it does some computation. Thus, should we name the getters `getCf()` and functions that compute something `computeCf` for example ? Also, most of them can be marked as [[nodiscard]] and need a docstring.
```suggestion
    // face centres
    std::span<const Vector> Cf(const int i) const;
```

Reply:
This naming is identical to OpenFOAM

Reply:
Ok lets rather keep `cf()` instead of having a `getCf(`), and lets not use [[nodiscard]] for getter. Also document the decision somewhere.